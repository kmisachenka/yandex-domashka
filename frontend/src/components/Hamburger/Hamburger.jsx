import React from 'react';
import * as PropTypes from 'prop-types';

import styles from './Hamburger.module.scss';
import { ReactComponent as HamburgerSVG } from './svg/hamburger.svg';

export default function Hamburger(props) {
  const { handleClick } = props;
  return (
    <button type="button" className={styles.button} onClick={handleClick}>
      <HamburgerSVG />
    </button>
  );
}

Hamburger.propTypes = {
  handleClick: PropTypes.func.isRequired,
};
