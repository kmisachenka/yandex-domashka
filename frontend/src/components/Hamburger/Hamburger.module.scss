.button {
  appearance: none;
  border: 0;
  margin: 0;
  padding: 5px;
}

.hamburger {
  background-color: #000;
  background-image: url('./svg/hamburger.svg');
  height: 16px;
  width: 22px;
}
