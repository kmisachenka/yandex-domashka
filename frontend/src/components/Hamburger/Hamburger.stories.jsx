import React from 'react';
import { storiesOf } from '@storybook/react';
import Hamburger from './Hamburger';

storiesOf('Гамбургер', module)
  .add('обычный', () => (
    <Hamburger />
  ));
