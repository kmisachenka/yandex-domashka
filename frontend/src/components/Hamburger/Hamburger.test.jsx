import React from 'react';
import { shallow, mount } from 'enzyme';
import Hamburger from './Hamburger';

describe('Hamburger', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Hamburger />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should handle click', () => {
    const handleClick = jest.fn();
    const wrapper = mount(<Hamburger handleClick={handleClick} />);
    const button = wrapper.find('.button');
    button.simulate('click');
    expect(handleClick).toHaveBeenCalled();
  });
});
