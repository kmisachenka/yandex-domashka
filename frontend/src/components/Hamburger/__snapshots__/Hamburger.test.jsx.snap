// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hamburger should match the snapshot 1`] = `
<button
  className="button"
  onClick={[Function]}
  type="button"
>
  <ForwardRef />
</button>
`;
